contracts/SPGNFT.sol

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.23;
+
+import { AccessControlUpgradeable } from "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import { ERC721Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+import { ISPGNFT } from "./interfaces/ISPGNFT.sol";
+import { Errors } from "./lib/Errors.sol";
+import { SPGNFTLib } from "./lib/SPGNFTLib.sol";
+
+contract SPGNFT is ISPGNFT, ERC721Upgradeable, AccessControlUpgradeable {
```

Comment:
Missing inherit from `UUPSUpgradeable`

Reply:
It's a Beacon proxy, like IPRoyaltyVault in the core repo, so no need for inheriting UUPS